map to map with filter cause NPE

When source and destination class have map field and additionally custom filter is defined mapping will cause NPE:

Caused by: ma.glasnost.orika.MappingException: While attempting the following mapping:
sourceType = MapToMapWithFilterTest.FirstClassWithMap
sourceProperty = map(Map<String, String>)
destinationType = MapToMapWithFilterTest.SecondClassWithMap
destinationProperty = map(Map<String, String>)
Error occurred: java.lang.NullPointerException
	at ma.glasnost.orika.impl.generator.MapperGenerator.addMapMethod(MapperGenerator.java:177)
	at ma.glasnost.orika.impl.generator.MapperGenerator.build(MapperGenerator.java:72)
	... 27 more
Caused by: java.lang.NullPointerException
	at ma.glasnost.orika.CustomFilter.appliesTo(CustomFilter.java:87)
	at ma.glasnost.orika.impl.generator.SourceCodeContext.getFilter(SourceCodeContext.java:939)
	at ma.glasnost.orika.impl.generator.SourceCodeContext.applyFilters(SourceCodeContext.java:815)
	at ma.glasnost.orika.impl.generator.SourceCodeContext.mapFields(SourceCodeContext.java:762)
	at ma.glasnost.orika.impl.generator.specification.MapToMap.generateMappingCode(MapToMap.java:81)
	at ma.glasnost.orika.impl.generator.SourceCodeContext.mapFields(SourceCodeContext.java:768)
	at ma.glasnost.orika.impl.generator.MapperGenerator.generateFieldMapCode(MapperGenerator.java:251)
	at ma.glasnost.orika.impl.generator.MapperGenerator.addMapMethod(MapperGenerator.java:174)
	... 28 more



-----
[MapToMapWithFilterTest.txt](https://github.com/orika-mapper/orika/files/1257266/MapToMapWithFilterTest.txt)

--- a/core/src/main/java/ma/glasnost/orika/CustomFilter.java
+++ b/core/src/main/java/ma/glasnost/orika/CustomFilter.java
@@ -18,11 +18,11 @@
 
 package ma.glasnost.orika;
 
-import java.lang.reflect.ParameterizedType;
-
 import ma.glasnost.orika.metadata.Property;
 import ma.glasnost.orika.metadata.Type;
 import ma.glasnost.orika.metadata.TypeFactory;
+
+import java.lang.reflect.ParameterizedType;
 
 /**
  * CustomFilter provides the base implementation of Filter.
@@ -84,8 +84,8 @@ public abstract class CustomFilter<A, B> implements Filter<A, B> {
      * @see ma.glasnost.orika.Filter#appliesTo(ma.glasnost.orika.metadata.Property, ma.glasnost.orika.metadata.Property)
      */
     public boolean appliesTo(Property source, Property destination) {
-        return sourceType.isAssignableFrom(getObjectType(source.getType())) &&
-               destinationType.isAssignableFrom(getObjectType(destination.getType()));
+        return source != null && sourceType.isAssignableFrom(getObjectType(source.getType())) &&
+                destination != null && destinationType.isAssignableFrom(getObjectType(destination.getType()));
     }
 
     private static Type<?> getObjectType(Type<?> type) {

--- a/tests/src/main/java/ma/glasnost/orika/test/community/Issue232Test.java
+++ b/tests/src/main/java/ma/glasnost/orika/test/community/Issue232Test.java
@@ -1,0 +1,64 @@
+package ma.glasnost.orika.test.community;
+
+import ma.glasnost.orika.MapperFactory;
+import ma.glasnost.orika.NullFilter;
+import ma.glasnost.orika.impl.DefaultMapperFactory;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class Issue232Test {
+
+    private MapperFactory mapperFactory;
+
+    @Before
+    public void setUp() throws Exception {
+        mapperFactory = new DefaultMapperFactory.Builder().build();
+        mapperFactory.registerFilter(new NullFilter<Map, Map>());
+        mapperFactory.classMap(FirstClassWithMap.class, SecondClassWithMap.class).byDefault().register();
+    }
+
+    @Test
+    public void test_map_to_map_with_filter() {
+        FirstClassWithMap firstMap = new FirstClassWithMap();
+        firstMap.setMap(new HashMap<String, String>());
+        firstMap.getMap().put("A", "1");
+
+        SecondClassWithMap transformed = mapperFactory.getMapperFacade().map(firstMap, SecondClassWithMap.class);
+
+        assertNotNull(transformed);
+        assertNotNull(transformed.getMap());
+        assertEquals("1", transformed.getMap().get("A"));
+    }
+
+    public static class FirstClassWithMap {
+
+        private Map<String, String> map;
+
+        public Map<String, String> getMap() {
+            return map;
+        }
+
+        public void setMap(Map<String, String> map) {
+            this.map = map;
+        }
+    }
+
+    public static class SecondClassWithMap {
+
+        private Map<String, String> map;
+
+        public Map<String, String> getMap() {
+            return map;
+        }
+
+        public void setMap(Map<String, String> map) {
+            this.map = map;
+        }
+    }
+}

